.github/workflows/integration-tests-v1.yml

```diff
@@ -44,25 +44,15 @@ jobs:
           python -m pip install --upgrade pip
           pip install ".[dev]" --no-cache-dir
 
-      - name: Determine env variables
-        run: |
-          if [ "${{ inputs.environment }}" == 'staging' ]; then
-             echo "USERNAME=${{ secrets.FIREBOLT_USERNAME_STAGING }}" >> "$GITHUB_ENV"
-             echo "PASSWORD=${{ secrets.FIREBOLT_PASSWORD_STAGING }}" >> "$GITHUB_ENV"
-          else
-             echo "USERNAME=${{ secrets.FIREBOLT_USERNAME_DEV }}" >> "$GITHUB_ENV"
-             echo "PASSWORD=${{ secrets.FIREBOLT_PASSWORD_DEV }}" >> "$GITHUB_ENV"
-          fi
-
       - name: Keep environment name in the summary
         run: echo '### Ran integration tests against ${{ inputs.environment }} ' >> $GITHUB_STEP_SUMMARY
 
       - name: Setup database and engine
         id: setup
         uses: firebolt-db/integration-testing-setup@v1
         with:
-          firebolt-username: ${{ env.USERNAME }}
-          firebolt-password: ${{ env.PASSWORD }}
+          firebolt-username: ${{ inputs.environment == 'staging' && secrets.FIREBOLT_STG_USERNAME || secrets.FIREBOLT_USERNAME_DEV }}
```

Comment:
Does this change anything?

Reply:
Yes, special characters were bugged in the previous implementation - https://github.com/firebolt-db/dbt-firebolt/pull/118/files#diff-29ce8407d15551c9c62184e207b6ceadebb261f7ffbfb75d7d67844ba33709abL51

if you had a $ in the password for example you'd have bash resolving that variable to an empty string therefore rendering the password unusable. This implementation does not pass around secrets but uses the correct one instead so it's kept as is.

Reply:
This is probably a bug in every repo. Yet, since we are removing dev support this will be fixed by it's own, so nothing to do I guess